src/gbm/gblinear_model.cc

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright 2019 by Contributors
+ */
+#include <utility>
+#include <limits>
+
+#include "gblinear_model.h"
+
+namespace xgboost {
+namespace gbm {
+
+void GBLinearModel::Save(Json* p_out) const {
+  using WT = std::remove_reference<decltype(std::declval<decltype(weight)>().back())>::type;
+  using JT = Number::Float;
+  static_assert(std::is_same<WT, JT>::value, "");
```

Comment:
Optional. What do you think about adding an error message: Weight and JT (what is that?) are supposed to be of the same type
or something like this :)